### ISSUE TITLE
nullPointer on close button with P3D and noLoop

### ISSUE THREAD

**Issue by hype** [processing/processing#5214]
```
void setup() {
 size(600,600,P3D);
 noLoop();
}

void draw() { 
}
```

OSX Sierra 10.12.6 / Processing 3.3.5

if you press the esc key = no problem
if you mouse click "x" close button = 
```
java.lang.NullPointerException
	at processing.core.PApplet.dispose(PApplet.java:3706)
	at processing.core.PApplet.exit(PApplet.java:3653)
	at processing.opengl.PSurfaceJOGL$NEWTWindowListener.windowDestroyNotify(PSurfaceJOGL.java:911)
	at jogamp.newt.WindowImpl.consumeWindowEvent(WindowImpl.java:4389)
	at jogamp.newt.WindowImpl.sendWindowEvent(WindowImpl.java:4317)
	at jogamp.newt.WindowImpl.windowDestroyNotify(WindowImpl.java:4677)
```
ONLY occurs when `noLoop()` is present
ONLY occurs in P3D

**Comment by christianbender** [comment]
I don't use OSX Sierra 10.12.6 but I use Linux Mint 18.2 and Processing 3.3.6.
If I close the window with 'ESC' or click of 'X'.  I get the message, below.
![screenshot1](https://user-images.githubusercontent.com/23243382/33137780-2acf056c-cfa9-11e7-9f91-676e81ca9513.png)

**Comment by gwygonik** [comment]
I have the same issue on OSX (Sierra 10.12.6) with Processing **3.3.6** but with _either_ P2D or P3D.

> java.lang.NullPointerException
> 	at processing.core.PApplet.dispose(PApplet.java:3718)
> 	at processing.core.PApplet.exit(PApplet.java:3665)
> 	at processing.opengl.PSurfaceJOGL$NEWTWindowListener.windowDestroyNotify(PSurfaceJOGL.java:921)
> 	at jogamp.newt.WindowImpl.consumeWindowEvent(WindowImpl.java:4389)
> 	at jogamp.newt.WindowImpl.sendWindowEvent(WindowImpl.java:4317)
> 	at jogamp.newt.WindowImpl.windowDestroyNotify(WindowImpl.java:4677)
>

**Comment by benfry** [comment]
Fixed by @hazmatsuitor for 3.3.7 or 3.4 (whatever we call the next release. 

@christianbender That looks like a different issue that can be filed separately. Thanks!
